grakn-kb/src/main/java/ai/grakn/kb/internal/concept/TypeImpl.java

```diff
@@ -241,7 +248,9 @@ boolean deletionAllowed(){
 
     Stream<V> instancesDirect(){
         return vertex().getEdgesOfType(Direction.IN, Schema.EdgeLabel.SHARD).
-                map(edge -> vertex().tx().factory().buildShard(edge.source())).
+                map(EdgeElement::source).
+                flatMap(CommonUtil::optionalToStream).
+                map(source -> vertex().tx().factory().buildShard(source)).
```

Comment:
I _think_ these kinds of lambdas can be made into method references like this:
```
map(vertex().tx().factory()::buildShard)
```
(although remember `vertex().tx().factory()` would now be called outside the lambda)

Reply:
It's can't in this case because you passing in `edge.source()` not `edge`